python/hsml/resources.py

```diff
@@ -181,59 +187,76 @@ def _get_default_resource_limits(self):
-                + str(max_resources["gpus"])
-                + " gpus."
-            )
+        if max_resources["cores"] > -1:
+            if limits.cores <= 0:
+                raise ValueError("Limit number of cores must be greater than 0 cores.")
+            if limits.cores > max_resources["cores"]:
+                raise ValueError(
+                    "Limit number of cores cannot exceed the maximum of "
+                    + str(max_resources["cores"])
+                    + " cores."
+                )
+        if max_resources["memory"] > -1:
+            if limits.memory <= 0:
+                raise ValueError("Limit memory resources must be greater than 0 MB.")
+            if limits.memory > max_resources["memory"]:
+                raise ValueError(
+                    "Limit memory resources cannot exceed the maximum of "
+                    + str(max_resources["memory"])
+                    + " MB."
+                )
+        if max_resources["gpus"] > -1:
+            if limits.gpus < 0:
+                raise ValueError(
+                    "Limit number of gpus must be greater than or equal to 0 gpus."
+                )
+            if limits.gpus > max_resources["gpus"]:
+                raise ValueError(
+                    "Limit number of gpus cannot exceed the maximum of "
+                    + str(max_resources["gpus"])
+                    + " gpus."
+                )
 
         # requests
-        if self._limits.cores > -1 and (
-            self._requests.cores < 0 or self._requests.cores > self._limits.cores
-        ):
+        if requests.cores <= 0:
+            raise ValueError("Requested number of cores must be greater than 0 cores.")
+        if limits.cores > -1 and requests.cores > limits.cores:
             raise ValueError(
                 "Requested number of cores cannot exceed the limit of "
-                + str(self._limits.cores)
+                + str(limits.cores)
                 + " cores."
             )
-        if self._limits.memory > -1 and (
-            self._requests.memory < 0 or self._requests.memory > self._limits.memory
-        ):
+        if requests.memory <= 0:
+            raise ValueError("Requested memory resources must be greater than 0 MB.")
+        if limits.memory > -1 and requests.memory > limits.memory:
             raise ValueError(
                 "Requested memory resources cannot exceed the limit of "
-                + str(self._limits.memory)
+                + str(limits.memory)
                 + " MB."
             )
-        if self._limits.gpus > -1 and (
-            self._requests.gpus < 0 or self._requests.gpus > self._limits.gpus
-        ):
+        if requests.gpus < 0:
+            raise ValueError(
+                "Requested number of gpus must be greater than or equal to 0 gpus."
+            )
+        if limits.gpus > -1 and requests.gpus > limits.gpus:
             raise ValueError(
```

Comment:
We saw before that requests.gpus and limits.gpus must always be equal. Are we adding that check in a separate PR?

Reply:
Yes, I'm working on a different PR for that.